joplin/base/views/joplin_search_views.py

```diff
@@ -0,0 +1,119 @@
+from django.shortcuts import render
+from django.http.request import QueryDict
+from django.contrib.contenttypes.models import ContentType
+from django.core.paginator import Paginator
+from wagtail.core.models import Page
+from wagtail.search.query import MATCH_ALL
+from wagtail.admin.forms.search import SearchForm
+
+def search(request):
+    pages = all_pages = Page.objects.all().prefetch_related('content_type').specific()
+    q = MATCH_ALL
+    content_types = []
+    pagination_query_params = QueryDict({}, mutable=True)
+    ordering = None
+
+    if 'ordering' in request.GET:
+        if request.GET['ordering'] in ['content_type', '-content_type', 'owner', '-owner', 'title', '-title', 'latest_revision_created_at', '-latest_revision_created_at', 'live', '-live']:
+            ordering = request.GET['ordering']
+
+            # The content_type order column was added so Joplin search results could order alphabetically by content_type
+            if ordering == 'content_type':
+                pages = pages.order_by('content_type')
+            elif ordering == '-content_type':
+                pages = pages.order_by('-content_type')
+
+            # The owner order column was added so Joplin search results could order alphabetically by owner
+            if ordering == 'owner':
+                pages = pages.order_by('owner')
+            elif ordering == '-owner':
+                pages = pages.order_by('-owner')
+
+            if ordering == 'title':
+                pages = pages.order_by('title')
+            elif ordering == '-title':
+                pages = pages.order_by('-title')
+
+            if ordering == 'latest_revision_created_at':
+                pages = pages.order_by('latest_revision_created_at')
+            elif ordering == '-latest_revision_created_at':
+                pages = pages.order_by('-latest_revision_created_at')
+
+            if ordering == 'live':
+                pages = pages.order_by('live')
+            elif ordering == '-live':
+                pages = pages.order_by('-live')
+
+    if 'content_type' in request.GET:
+        pagination_query_params['content_type'] = request.GET['content_type']
+
+        app_label, model_name = request.GET['content_type'].split('.')
+
+        try:
+            selected_content_type = ContentType.objects.get_by_natural_key(app_label, model_name)
+        except ContentType.DoesNotExist:
+            raise Http404
+
+        pages = pages.filter(content_type=selected_content_type)
+    else:
+        selected_content_type = None
+
+    query = ''
+
+    # JOPLIN NOTE: Some of this the original state of the query condition
```

Comment:
multiline comments are much easier to read/write as docstrings

Reply:
I've added those in the updated PR